Warn when automatically lowering an inductive declared with `: Type` to Prop (#18989)
coq/coq: doc/sphinx/language/core/inductive.rst

      Inductive nat : Set := O | S (_:nat).

Automatic Prop lowering
+++++++++++++++++++++++

> When an inductive is declared with no explicit sort, it is put in the
> smallest sort which does not prevent large elimination (excluding

Reviewer: ```suggestion
When an inductive is declared without an explicit sort, it is put in the
smallest sort which permits large elimination (excluding
```

`SProp`). For :ref:`empty and singleton <Empty-and-singleton-elimination>`
types this means they are declared in `Prop`.

.. flag:: Automatic Proposition Inductives

   By default the above behaviour is extended to empty and singleton
   inductives explicitly declared in `Type` (but not those in explicit
   universes using `Type@{u}`, or in `Type` through an auxiliary definition
   such as `Definition typ := Type.`).

   Disabling this flag prevents inductives with an explicit non-`Prop`
   type from being lowered to `Prop`. This will become the default in

>    a future version. Note :flag:`Dependent Proposition Eliminators` to

Reviewer: ```suggestion
   a future version. Use :flag:`Dependent Proposition Eliminators` to
```
